### PR TITLE
NF-95 - Entry details page

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/EntryDetailsPage.scala.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        inputDate: components.inputDate,
+        errorSummary: errorSummary,
+        continueButton: components.continueButton )
+
+@(form: Form[EntryDetails])(implicit request: Request[_], messages: Messages)
+
+@layout(pageTitle = Title(form, messages("entryDetails.title"))) {
+
+    @formHelper(action = EntryDetailsController.onSubmit(), 'autoComplete -> "off") {
+        @errorSummary(form.errors, focusOverrides = Map("entryDate" -> "entryDate_day"))
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/EntryDetailsPage.scala.html:39]: OBSERVATION: We could potentially remove the need for `focusOverrides` (and the markup having knowledge of specific IDs) _if_ the `@inputDate` component gave the day input an ID equal to the `fieldName`

File: build.sbt [hunk 2]
```diff
@@ -51,7 +51,7 @@ lazy val scoverageSettings: Seq[Setting[_]] = Seq(
     ".*(BuildInfo|Routes|Options|LanguageSwitchController|LanguageSelect).*",
     "logger.*\\(.*\\)"
   ).mkString(";"),
-  coverageMinimum := 87.5,
+  coverageMinimum := 90,
```
Comment (crosslandwa) [inline build.sbt:54]: This number keeps going up ! 👍

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/forms/mappings/MappingsSpec.scala [hunk 3]
```diff
@@ -155,4 +157,90 @@ class MappingsSpec extends UnitSpec with OptionValues with Mappings {
       result.errors must contain(FormError("value", "error.required"))
     }
   }
+
+  "localData" must {
```
Comment (crosslandwa) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/forms/mappings/MappingsSpec.scala:161]:
```suggestion
  "localDate" must {
```

**Review by crosslandwa — DISMISSED** [review]

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/EntryDetailsPage.scala.html (hunk 1 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/EntryDetailsPage.scala.html]: Took some doing but I've removed this feature now.  It did involve a compromise that the `inputDate` component has to make assumptions about the format of error message keys for it to correctly highlight multiple fields in error but at least now the consumer of the  component doesn't have to do anything special for it all to work.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/EntryDetailsController.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim
+
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.actions.IdentifierAction
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.forms.{BankDetailsFormProvider, EntryDetailsFormProvider}
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/EntryDetailsController.scala:23]: Unused import

```suggestion
import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.forms.EntryDetailsFormProvider
```

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/EntryDetailsController.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.navigation.Navigator
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.pages.{BankDetailsPage, EntryDetailsPage}
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/EntryDetailsController.scala:25]: Unused import

```suggestion
import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.pages.EntryDetailsPage
```

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**